bindata/network/ovn-kubernetes/alert-rules-control-plane.yaml

```diff
@@ -15,9 +15,9 @@ spec:
     - alert: NoRunningOvnMaster
       annotations:
         message: |
-          there is no running ovn-kubernetes master
+          There is no running ovn-kubernetes master
       expr: |
-        absent(up{job="ovnkube-master",namespace="openshift-ovn-kubernetes"} ==1)
+        absent(up{job="ovnkube-master-metrics", namespace="openshift-ovn-kubernetes"} == 1)
```

Comment:
This can stay as is. The only thing we need is the `app: ovnkube-master` on the `ovnkube-db` service. Which it already does. So the only real change needed is in the `ServiceMonitor` where the `matchLabels` stanza should have `app: ovnkube-master`?

Reply:
> This can stay as is. The only thing we need is the `app: ovnkube-master` on the `ovnkube-db` service. Which it already does. So the only real change needed is in the `ServiceMonitor` where the `matchLabels` stanza should have `app: ovnkube-master`?

umm no, I mean if we do that - as in use the `ovnkube-db` service then we have to expose the metrics port on the `ovnkube-db` service so that prometheus can scrape that endpoint which basically defeats the purpose of having a `ovnkube-master-metrics` service.

Reply:
I am thoroughly confused. The selector for the "app" it monitors is orthogonal to the port(s) it monitors on. 

```
endpoints:
  - interval: 30s
    port: metrics
    bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
    scheme: https
    tlsConfig:
      caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
      serverName: ovn-kubernetes-master-metrics.openshift-ovn-kubernetes.svc
``` 
is telling where to get the metrics from. The 

```
  selector:
    matchLabels:
      app: ovnkube-master
``` 
is telling what the "job" is. So technically you can still setup the endpoint to get the metrics from to be the port (`metrics`) specified on the [ovnkube-master-metrics service ](https://github.com/openshift/cluster-network-operator/blob/master/bindata/network/ovn-kubernetes/monitor.yaml#L28) and have that check for pods that have `app:ovnkube-master` label. No? Is my understanding incorrect?

Reply:
So Juan and I tested this, and we had to change the labels on the `ovn-kubernetes-master-metrics` service to match the master pods labels for the metric to be pulled from master pods. So we use the `ovn-kubernetes-master-metrics` instead of `ovnkube-db`